engine/core/src/main/java/org/datacleaner/metamodel/datahub/DatahubDataContext.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.metamodel.datahub;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.security.AccessControlException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.metamodel.AbstractDataContext;
+import org.apache.metamodel.UpdateScript;
+import org.apache.metamodel.UpdateableDataContext;
+import org.apache.metamodel.data.DataSet;
+import org.apache.metamodel.query.Query;
+import org.apache.metamodel.schema.Schema;
+import org.datacleaner.metamodel.datahub.utils.JsonSchemasResponseParser;
+import org.datacleaner.util.http.MonitorHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DatahubDataContext extends AbstractDataContext implements
+        UpdateableDataContext {
+    private static final Logger logger = LoggerFactory
+            .getLogger(DatahubDataContext.class);
+
+    private DatahubConnection _connection;
+    private Map<String, DatahubSchema> _schemas;
+
+    public DatahubDataContext(DatahubConnection connection) {
+        _connection = connection;
+        _schemas = getDatahubSchemas();
+    }
+
+    private static final Pattern COLON = Pattern
+            .compile("%3A", Pattern.LITERAL);
+    private static final Pattern SLASH = Pattern
+            .compile("%2F", Pattern.LITERAL);
+    private static final Pattern QUEST_MARK = Pattern.compile("%3F",
+            Pattern.LITERAL);
+    private static final Pattern EQUAL = Pattern
+            .compile("%3D", Pattern.LITERAL);
+    private static final Pattern AMP = Pattern.compile("%26", Pattern.LITERAL);
+
+    public static String encodeUrl(String url) {
```

Comment:
I can recommend Guava's UrlEscapers class. It has all this and then some. Probably more safe to use.